src/components/AuthRoute.js

```diff
@@ -93,6 +87,18 @@ class AuthRoute extends React.Component<AuthRouteDefaultProps, AuthRouteProps, A
         />
       );
     }
+
+    // if authentication is not required or is handled outside this application,
+    // then only check that the user is logged in if specific credentials are required for our route
+    if (!this.props.required || AuthUtils.isLoggedIn(this.props.required)) {
+      return (
+        <Route
+          {...this.props}
+        />
+      );
+    }
+
+    // insufficient credentials (provided by external authentication system) => don't let the user through
```

Comment:
Maybe "If no particular authentication is required or one is and the user has it, show the route."